### PR TITLE
docs: Adding documentation

### PR DESCRIPTION
**Pull request by DenverCoder1**

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+rst_prolog = """
+.. |coro| replace:: This function is a |coroutine_link|_.
+.. |maybecoro| replace:: This function *could be a* |coroutine_link|_.
+.. |coroutine_link| replace:: *coroutine*
+.. _coroutine_link: https://docs.python.org/3/library/asyncio-task.html#coroutine
+"""
+
+highlight_language = "python3"
+html_theme = "sphinx_rtd_theme"
```
Comment (VincentRPS) [inline docs/conf.py:70]: I would suggest to use the Sphinx book theme imo it looks better

File: docs/ext/menus/api.rst [hunk 2]
```diff
@@ -0,0 +1,125 @@
+API Reference
+=============
+
```
Comment (VincentRPS) [inline docs/ext/menus/api.rst:3]:
```suggestion

.. _ext_menus_api:

```

File: docs/ext/menus/button_menus.rst [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Using Button Menus
+==================
+
+.. currentmodule:: nextcord.ext.menus
+
```
Comment (VincentRPS) [inline docs/ext/menus/button_menus.rst:5]:
```suggestion

.. _ext_menus_button_menus:
```

File: docs/ext/menus/reaction_menus.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Using Reaction Menus
+====================
+
```
Comment (VincentRPS) [inline docs/ext/menus/reaction_menus.rst:3]:
```suggestion

.. _ext_menus_reaction_menus:
```

**Review by VincentRPS — CHANGES_REQUESTED** [review]

**Comment by parker02311** [comment]
Documentation is a great idea, I was actually gonna work on this today. This will make understanding menus much easier for newer users.

**Comment by parker02311** [comment]
LGTM

File: docs/conf.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
+import re
+import sys
+import os
+
+
+project = "nextcord-ext-menus"
+copyright = "2021-present, Nextcord. 2015-2020, Danny (Rapptz)"
```
Comment (VincentRPS) [inline docs/conf.py:7]: Fixed incorrect copyright years
```suggestion
copyright = "2021-present, Nextcord. 2020-2021, Danny (Rapptz)"
```

**Review by VincentRPS — CHANGES_REQUESTED** [review]
Fix the copyright year

**Review by parker02311 — APPROVED** [review]
LGTM
Should push ASAP for those that need help understanding menus

**Pull request merged**